### ISSUE TITLE
Search should consider iskeyword characters for word boundaries

### ISSUE THREAD

**Issue by chessami92** [vrapper/vrapper#739]
In my vrapperrc I have changed iskeyword:
`set iskeyword-=_`
The asterisk seems to correctly identify words bound by '_' but the search function still counts underscores as keywords. That is, if I press asterisk on the left side of "TESTING_123" the search pattern is "\\\<TESTING\\\>" but even the current word under the cursor cannot be found with this pattern.

**Comment by albertdev** [comment]
We use the Eclipse / Java regex search engine under the covers.

While our keyword-detection logic decently works for grabbing the current keyword, Eclipse is unaware of such fancy word boundaries. I'm going to mark this issue as a feature request then, searching for that exact same pattern is simply not supported today.

**Comment by albertdev** [comment]
We use the Eclipse / Java regex search engine under the covers.

While our keyword-detection logic decently works for grabbing the current keyword, Eclipse is unaware of such fancy word boundaries. I'm going to mark this issue as a feature request then, searching for that exact same pattern is simply not supported today.

**Comment by albertdev** [comment]
I changed the title as the previous one sounded ambiguous to me.

**Comment by keforbes** [comment]
I actually can't reproduce this one.  I have `hlsearch` enabled and I hit `*` on `TESTING` to show what's matching with each scenario.  I'm also bringing up the command-line and hitting `up` to show the history of last command entered:
![one](https://cloud.githubusercontent.com/assets/817509/14824392/088f7684-0b93-11e6-8ca3-b814aee8fa80.png)


![two](https://cloud.githubusercontent.com/assets/817509/14824397/0da9f310-0b93-11e6-902e-54a3c4ad7928.png)

If I then look in my search history (hit `/` then `up`), things appear to be set correctly:
![three](https://cloud.githubusercontent.com/assets/817509/14824450/3e4866b4-0b93-11e6-8142-81d9cd221206.png)

![four](https://cloud.githubusercontent.com/assets/817509/14824455/418e8f88-0b93-11e6-9798-39a1bc1c2c75.png)

What am I missing?  Am I doing something differently?

**Comment by keforbes** [comment]
Bah, of course, it's only after I post a comment that I understand what you're saying.  My last screenshot is what's wrong.  When searching for `\<testing\>`, Vrapper should treat `_` as a word boundary and therefore `\>` should match it.

![five](https://cloud.githubusercontent.com/assets/817509/14824762/5b1d2094-0b94-11e6-9600-6733104a1163.png)

  @albertdev was right, that's all in the `wholeWord` boolean that we pass to Eclipse's search engine.  I wonder if we can set the `wholeWord` boolean to false and use the Java regex whole word `\b` instead so we can modify our regex to essentially `(\b)|(keyword)` when hitting `*`.  Of course, we'd still have to display `\>` to the user so this would all have to happen under the hood.  And now I'm just rambling.

**Comment by chessami92** [comment]
Creating a full regex with \b while showing and accepting \< \> notation is the implementation I was thinking of as well.

**Comment by keforbes** [comment]
I'm sorry for just rambling more here...

As a test, I tried hardcoding the `wholeWord` boolean to `false` to see what the behavior was.  As far as I can tell, it worked perfectly.  So I looked through the code to see if I was affecting any other features.  Again, unless I'm missing something, literally every call to search (other than `*` and `#`) either hardcodes `wholeWord` to `false` or takes whatever `wholeWord` value was passed in (which was `false`).  This can't be the solution, but I'm not sure what I'm missing.  I think the main thing here is that we define what `*` means and we've already found the search term based on our `iskeyword` setting; the eclipse search engine doesn't need to know that we determined the search term based on word boundaries.

I think there's only one scenario that this wouldn't handle: users manually entering `\>` into their search.  I'm already replacing all `\>` with Java's `\b` under the hood so I should be able to change that to my `(\b)|(keywords)` crazy idea from earlier.  This may be possible...
